fe1-web/model/objects/WalletCryptographyHandler.ts

```diff
@@ -0,0 +1,76 @@
+import { IndexedDBStore, WalletCryptoKey } from 'store/stores/IndexedDBStore';
+import STRINGS from 'res/strings';
+
+/**
+ * This class has the job of handling the cryptography functions of the wallet.
+ * It interacts with the IndexedDB database in order to store and retrieve the
+ * secret key. It also encrypts and decrypts the tokens with the retrieved key.
+ */
+export class WalletCryptographyHandler {
+  /* encryption/decryption algorithm (RSA) */
+  private readonly algorithm = {
+    name: 'RSA-OAEP',
+    modulusLength: 4096,
+    publicExponent: new Uint8Array([1, 0, 1]),
+    hash: 'SHA-256',
+  };
+
+  private readonly keyUsages: KeyUsage[] = ['encrypt', 'decrypt'];
+
+  /* the encryption/decryption key storage */
+  private keyDatabase: IndexedDBStore = new IndexedDBStore();
+
+  /**
+   * this method adds an RSA encryption/decryption key,
+   * interacts with IndexedDB browser database.
+   */
+  public async initWalletStorage(): Promise <void> {
+    const key: WalletCryptoKey = await this.generateRSAKey();
```

Comment:
this means that every time you initialize the wallet, you're going to overwrite the key? That seems very dangerous...